### PR TITLE
Define unzip and unzip3 for Option

### PR DESCRIPTION
**Pull request by neontorrent**
Fix scala/bug#11222

Related issue: scala/bug#8394
Related PR: [scala/pull/6491](https://github.com/scala/scala/pull/6491)

### CODE REVIEW

**Comment by texasbruce** [comment]
cla has been signed

File: src/library/scala/Option.scala [hunk 1]
```diff
@@ -323,6 +323,43 @@ sealed abstract class Option[+A] extends Product with Serializable {
...
+  /** Converts an Option of pairs into an Option of first elements and an Option of second elements.
+    *
+    *  @tparam A1    the type of the first half of the element pairs
+    *  @tparam A2    the type of the second half of the element pairs
+    *  @param asPair an implicit conversion which asserts that the element type
+    *                of this Option is a pair.
+    *  @return       a pair of Options, containing, respectively, the first and second half
+    *                of each element pair of this Option.
+    */
+  final def unzip[A1, A2](implicit asPair: A => (A1, A2)): (Option[A1], Option[A2]) = {
```
Comment (hrhino) [inline src/library/scala/Option.scala:335]: should this be `A <:< (A1, A2)` instead? I'm not sure how much I like using a view for this.

**Comment by SethTisue** [comment]
thank you for the ticket and pull request — I think it was just an oversight that this wasn't added earlier, glad to have it addressed.

(we are now being very conservative about accepting any further new API for 2.13, but this seems like sufficiently modest and uncontroversial)

I think the null handling you have is not as it should be. in Scala, `Some(null)` is considered legal and valid. the only place we handle `null` specially is `Option.apply` (and we're even considering whether that ought to happen without being signaled in the method name, see https://github.com/scala/scala/pull/7336), the rest of the `Option` API allows nulls through, `unzip` should too, IMO.

File: src/library/scala/Option.scala (hunk 1 above)
Comment (neontorrent) [reply src/library/scala/Option.scala:335]: This is the same signature as other container such as Array.unzip They should be aligned I think.

Comment (hrhino) [reply src/library/scala/Option.scala:335]: That's true. 

Is there a reason that the other `unzip`s use views rather than `<:<`, I wonder?

Comment (neontorrent) [reply src/library/scala/Option.scala:335]: That will be out of my knowledge. Maybe open a new discussion?

**Comment by neontorrent** [comment]
@SethTisue Make sense. Other unzip such as Array((null)).unzip also returns Some(null).

I changed to using Some. Please review.

File: src/library/scala/Option.scala (hunk 1 above)
Comment (hrhino) [reply src/library/scala/Option.scala:335]: Yep, sorry; I was asking in hopes that someone with more knowledge than I would know...

**Comment by texasbruce** [comment]
I squash the wrong commit. Will create a new PR - closing this one

**Pull request closed without merging**